gson/test/org/immutables/gson/bridge/GsonJacksonBridgeSerializationTest.java

```diff
@@ -0,0 +1,247 @@
+            testObject
+            , TestObject.class
+            , out -> {
+              try {
+                return factory.createGenerator(out);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , os -> {
+              try {
+                return factory.createParser(os.toByteArray());
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , JsonParserReader::new);
+  }
+
+  @Test
+  public void smileFactoryTest() throws IOException {
+    TestObject testObject = createTestObject();
+    SmileFactory factory = new SmileFactory();
+    runTestOnFactory(
+            testObject
+            , TestObject.class
+            , out -> {
+              try {
+                return factory.createGenerator(out);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , os -> {
+              try {
+                return factory.createParser(os.toByteArray());
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , JsonParserReader::new);
+  }
+
+  @Test
+  public void propertiesFactoryTest() throws IOException {
+    TestObject testObject = ImmutableTestObject.copyOf(createTestObject())
+            // excluding map here because when it is de-serialized, the order is changed and the unit test fails
+            .withMapOfStringToString(new HashMap<>());
+    JavaPropsFactory factory = new JavaPropsFactory();
+    runTestOnFactory(
+            testObject
+            , TestObject.class
+            , out -> {
+              try {
+                return factory.createGenerator(out);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , os -> {
+              try {
+                return factory.createParser(os.toByteArray());
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , XmlParserReader::new);
+  }
+
+  @Test
+  public void cborFactoryTest() throws IOException {
+    TestObject testObject = createTestObject();
+    CBORFactory factory = new CBORFactory();
+    runTestOnFactory(
+            testObject
+            , TestObject.class
+            , out -> {
+              try {
+                return factory.createGenerator(out);
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , os -> {
+              try {
+                return factory.createParser(os.toByteArray());
+              } catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+            , JsonParserReader::new);
+  }
+
+  private <VALUE
```

Comment:
This method is hard to follow (5 arguments and 3 generic types). 
Do you need separate `WRITER` / `READER` generic types ? Can one keep them as subclasses to `JsonGenerator` / `JsonParser`

Reply:
Sure can re-write it.